lib/order.rb

```diff
@@ -0,0 +1,100 @@
+require 'csv'
+require_relative 'customer'
+class Order
+
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+
+    # compared the fulfillment status against the acceptable statuses
+    if %i[pending paid processing shipped complete].include? (fulfillment_status)
+      @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError, "#{fulfillment_status} is an invalid fulfillment status"
+    end
+  end
+
+  # calculate the total cost of the order
+  def total
+    # extracts all of the values from the hash and sums them up
+    cost = @products.values.sum
+    # add 7.5% tax and round to two decimal places
+    return (cost * 1.075).round(2)
+  end
+
+  def add_product(product_name, price)
+    # raise error if there is a duplicate product
+    if @products.keys.include?(product_name)
```

Comment:
This doesn't matter here but you should get in the habit of avoiding things like `keys.include?` this is because `.keys` generates an array and `Array`'s `include?` is an O(n) operation.

If you use `include?` on the `Hash` itself it's an O(1) operation:

```suggestion
    if @products.include?(product_name)
```